src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs

```diff
@@ -156,8 +162,212 @@ internal static async Task<GetQrCodeResult> GetQrCode(HttpClient httpClient, Uri
+        {
+            Validation.NotNullOrEmpty(receiptId, nameof(receiptId));
+            try
+            {
+                var receiptResponse = await GetReceipt(httpClient, receiptId, clientSdkId, apiUrl, key);
+                var itemKeyId = receiptResponse.WrappedItemKeyId;
+                
+                var encryptedItemKeyResponse = await GetReceiptItemKey(httpClient, itemKeyId, clientSdkId, apiUrl, key);
+
+                var wrappedKey = Encoding.ASCII.GetBytes(receiptResponse.WrappedKey);
+                var receiptContentKey = CryptoEngine.UnwrapReceiptKey(wrappedKey, encryptedItemKeyResponse.Value, encryptedItemKeyResponse.Iv, key);
+ 
+                var (attrData, aextra, decryptAttrDataError) = DecryptReceiptContent(receiptResponse.Content, receiptContentKey);
+                if (decryptAttrDataError != null)
+                {
+                    throw new Exception($"An unexpected error occurred: {decryptAttrDataError.Message}");
+                   
+                }
+
+                var parsedProfile = ParseProfileContent(key, receiptResponse.WrappedKey, receiptResponse.OtherPartyContent.Profile);
+                var userProfile = new YotiProfile(parsedProfile
+                );
+       
+                var applicationProfile = new ApplicationProfile(
+                    ParseProfileContent(key, receiptResponse.WrappedKey,  receiptResponse.Content.Profile));
+
+                ExtraData userExtraData = new ExtraData();
+                userExtraData = CryptoEngine.DecryptExtraData(receiptResponse.WrappedKey,
+                        receiptResponse.OtherPartyContent.ExtraData,
+                        key);
+
+                ExtraData appExtraData = new ExtraData();
+                appExtraData = CryptoEngine.DecryptExtraData(
+                        receiptResponse.WrappedKey,
+                        receiptResponse.Content.ExtraData,
+                        key);
+
+                var sharedReceiptResponse = new SharedReceiptResponse
+                {
+                    ID = receiptResponse.ID,
+                    SessionID = receiptResponse.SessionID,
+                    RememberMeID = receiptResponse.RememberMeID,
+                    ParentRememberMeID = receiptResponse.ParentRememberMeID,
+                    Timestamp = receiptResponse.Timestamp,
+                    UserContent = new UserContent
+                    {
+                        UserProfile = userProfile,
+                        ExtraData = userExtraData
+                    },
+                    ApplicationContent = new ApplicationContent
+                    {
+                        ApplicationProfile = applicationProfile,
+                        ExtraData = appExtraData
+                    },
+                    Error = receiptResponse.Error
+                };
```

Comment:
I don't know how it works in .NET but few properties are optional in the receipt (`rememberMeId`, `parentRememberMeId` for instance) make sure to set these values only if not null